internal/app/alertmanager/issuecache/cache.go

```diff
@@ -94,6 +94,82 @@ func (c *Cache) CreateAlert(ctx context.Context, alert smee.Alert) (smee.Issue,
 	return issue, nil
 }
 
+func (c *Cache) AcknowledgeAlertsForIssue(ctx context.Context, issueID string) (smee.Issue, error) {
+	c.issuesMu.Lock()
+	defer c.issuesMu.Unlock()
+
+	if c.IssueStore != nil {
+		iss, err := c.IssueStore.AcknowledgeAlertsForIssue(ctx, issueID)
+		if err != nil {
+			return smee.Issue{}, fmt.Errorf("unable to acknowledge issue on substore: %w", err)
+		}
+
+		if iss.Active() {
+			c.issues[iss.ID] = iss
+		} else {
+			delete(c.issues, iss.ID)
+		}
+
+		return iss, nil
+	}
+
+	issue, ok := c.issues[issueID]
+	if !ok {
+		return smee.Issue{}, errors.New("issue does not exist")
+	}
+
+	return issue, nil
+}
+
+//==================================================/
+
+func (c *Cache) UnacknowledgeIssue(ctx context.Context, issueID string) (smee.Issue, error) {
+	c.issuesMu.Lock()
+	defer c.issuesMu.Unlock()
+
+	if c.IssueStore != nil {
+		iss, err := c.IssueStore.UnacknowledgeIssue(ctx, issueID)
+		if err != nil {
+			return smee.Issue{}, fmt.Errorf("unable to unacknowledge issue: %w", err)
+		}
+		return iss, nil
+	}
+
```

Comment:
I think that we need to update the cache here if I'm not mistaken. We are un-acknowledging  the issue in the db but not updating the cache layer. We need to add something like the following:

```
		if iss.Active() {
			c.issues[iss.ID] = iss
		} else {
			delete(c.issues, iss.ID)
		}
```